File: archived/2023-01/845b.cpp
#include <bits/stdc++.h>
using namespace std;

// g++ sample.cpp -o run
// ./run < input.txt

int main() {
    int t, n;
    long long ans;
    cin >> t;
    vector<long long> ls(1e5 + 1);
    vector<long long> ls2(1e5 + 1);
    ls[1] = 1;
    ls2[1] = 1;
    long long mod = 1e9 + 7;
    for (int i=2; i<=1e5; i++) {
        ls[i] = (ls[i-1] * i) % mod;
        ls2[i] = (ls2[i-1] + ls[i]) % mod;
        // cout << ls[i] << " " << ls2[i] << " ";
    }
    // cout << endl;
    while (t--) {
        cin >> n;
        ans = (((ls2[n-1] * n) % mod * (n - 1)) % mod + ((ls[n] * n) % mod * (n - 1) / 2) % mod) % mod;
        cout << ans << endl;
    }
}

File: archived/2023-01/845b.py
# import math, heapq, bisect, itertools, functools
# from collections import deque, Counter, defaultdict, OrderedDict

# python sample.py < input.txt

if __name__ == '__main__':
    t = int(input())
    ls = [0] * int(1e5+1)
    mod = int(1e9+7)
    ls[1] = 1
    for i in range(2, int(1e5+1)):
        ls[i] = (ls[i-1] * i) % mod
    for _ in range(t):
        n = int(input())
        ans = ls[n] * n * (n - 1) % mod
        print(ans)